tests/msc4140/delayed_event_test.go

```diff
@@ -120,6 +120,55 @@ func TestDelayedEvents(t *testing.T) {
 			},
 		})
 	})
+
+	t.Run("delayed state events are kept on server restart", func(t *testing.T) {
```

Comment:
> It may be possible to emulate this in Synapse's unit tests by pre-populating the database before starting the homeserver(?).

I'd like to do this anyways, so Synapse can have its own version of this test.  I'm unsure of how though, so I figured using Complement was sufficient to test it for the time being.

> For now, I'd suggest just leaving this here, and skipping tests the test if the running homeserver implementation is something other than Synapse.

Good idea. Done in cd42448

Reply:
Thanks. It's a bit of a shame that we have to list all homeserver implementations, and don't just have a `runtime.SkipIfNot(...)`. But that's a matter for a new PR.